discovery_engine/lib/src/domain/event_handler.dart

```diff
@@ -213,126 +155,32 @@ class EventHandler {
-    final sourceReactedRepository = HiveSourceReactedRepository();
-    Future<void> clearAiState() async {
-      await documentRepository.box.clear();
-      await activeDataRepository.box.clear();
-      await activeSearchRepository.box.clear();
-      await engineStateRepository.box.clear();
-    }
-
-    final sourcePreferenceRepository = HiveSourcePreferenceRepository();
-
     final setupData = await _fetchAssets(config);
-    final engineState = cfgFeatureStorage
-        ? null // unused
-        : await engineStateRepository.load();
-    final history = cfgFeatureStorage
-        ? <HistoricDocument>[] // unused
-        : await documentRepository.fetchHistory();
-    final reactedSources = cfgFeatureStorage
-        ? <SourceReacted>[] // unused
-        : await sourceReactedRepository.fetchAll();
-    final trustedSources = cfgFeatureStorage
-        ? <Source>{} // unused
-        : await sourcePreferenceRepository.getTrusted();
-    final excludedSources = cfgFeatureStorage
-        ? <Source>{} // unused
-        : await sourcePreferenceRepository.getExcluded();
     final availableSources = config.isMocked()
         ? mockedAvailableSources
         : await setupData.getAvailableSources();
-
-    final dartMigrationData = cfgFeatureStorage
-        ? await DartMigrationData.fromRepositories(
-            engineStateRepository,
-            documentRepository,
-            activeSearchRepository,
-            activeDataRepository,
-            sourceReactedRepository,
-            sourcePreferenceRepository,
-          )
-        : null;
-
-    final Engine engine;
-    try {
-      engine = await _initializeEngine(
-        EngineInitializer(
-          config: config,
-          setupData: setupData,
-          engineState: engineState,
-          history: history,
-          reactedSources: reactedSources,
-          deConfig: deConfig,
-          trustedSources: trustedSources,
-          excludedSources: excludedSources,
-        ),
-        dartMigrationData,
-      );
-    } catch (e) {
-      if ('$e'.contains('Unsupported serialized data.')) {
-        await engineStateRepository.clear();
-        throw InvalidEngineStateException('$e');
-      }
-      rethrow;
-    }
-
-    await dartMigrationData?.cleanup();
-
-    // init managers
-    _documentManager = DocumentManager(
-      engine,
-      documentRepository,
-      activeDataRepository,
-      engineStateRepository,
-      _changedDocumentsReporter,
-      sourceReactedRepository,
-    );
-    _feedManager = FeedManager(
-      engine,
-      documentRepository,
-      activeDataRepository,
-      engineStateRepository,
-      sourceReactedRepository,
-      sourcePreferenceRepository,
-      availableSources,
-    );
-    _searchManager = SearchManager(
-      engine,
-      activeSearchRepository,
-      documentRepository,
-      activeDataRepository,
-      engineStateRepository,
+    final dartMigrationData = await DartMigrationData.fromRepositories(
```

Comment:
maybe it makes sense to rename it?

```suggestion
    final dartMigrationData = await DartMigrationData.fromDirectoryPath(
```